app.js

```diff
@@ -3,21 +3,33 @@ const app = express();
 app.use (express.json()) ;
 const {
     getTopics,
-    getData
+    getData,
+    getArticleById
 } = require('./controllers/controller.js');
 
 
 app.get('/api/topics', getTopics);
 app.get('/api', getData);
+app.get('/api/articles/:article_id', getArticleById);
+
+
+
+app.use((err, req, res, next) => {
+  if (err.status && err.msg) {
+    res.status(err.status).send({ msg: err.msg });
+  }
+  else if (err.code === "23502" || err.code === "22P02" || "Invalid query") {
+    res.status(400).send({msg: "Bad request"});
+  }
+})
```

Comment:
This error handler has no `else` case so will trap any errors that come into it that don't match your `if` statements.

You need to call next with the error otherwise you'll never get to your 500 handler 👍